lint/type-guard_checker.go

```diff
@@ -18,6 +23,67 @@ func NewTypeGuardChecker(ctx *Context) *TypeGuardChecker {
 
 // Check runs type switch checks for f.
 func (c *TypeGuardChecker) Check(f *ast.File) []Warning {
-	// TODO: implement me. Refs #10.
-	return nil
+	c.warnings = c.warnings[:0]
+	// TODO(quasilyte): do recursive analysis. Type switches
+	// may have other type switches inside their clauses that
+	// can trigger warnings as well.
+	for _, decl := range collectFuncDecls(f) {
+		ast.Inspect(decl.Body, func(x ast.Node) bool {
+			if stmt, ok := x.(*ast.TypeSwitchStmt); ok {
+				c.check(stmt)
+				return false
+			}
+			return x != nil
```

Comment:
It doesn't really necessary. Inspect will never follow to nil child.
```
If f returns true, Inspect invokes f recursively 
for each of the non-nil children of node,
followed by a call of f(nil).
```
Also we might neet to check decl.Body for nil befor calling Inspect. Bc inspect shouldn't be invoked with nil argument.
``` Body *BlockStmt    // function body; or nil for external (non-Go) function```

Reply:
We need to replace `ast.Inspect` because it doesn't terminate traversal even after `false` is returned. But you're right. `x != nil` is not needed. Will fix that.

Good point about `nil` function bodies, thanks.